cynic-parser/src/parser/schema.lalrpop

```diff
@@ -43,8 +43,14 @@ TypeSystemDefinition: DefinitionId = {
 }
 
 SchemaDefinition: SchemaDefinitionRecord = {
-    schema "{" <root_operations:RootOperationTypeDefinition*> "}" => SchemaDefinitionRecord {
+    schema <directives:Directives> => SchemaDefinitionRecord {
         description: None,
+        directives,
+        root_operations: ast.no_root_operations(),
+    },
+    schema <directives:Directives> "{" <root_operations:RootOperationTypeDefinition*> "}" => SchemaDefinitionRecord {
+        description: None,
+        directives,
```

Comment:
Is there a better way to make the block optional?

Reply:
You could probably extract out a `RootOperationTypeDefinitions` like I've done with `FieldDefinitions` here: https://github.com/obmarg/cynic/blob/de668ef25ef1302b9f542d9f6a20463b07941752/cynic-parser/src/parser/schema.lalrpop#L118-L120

Reply:
I went that way, that seems to work nicely.